src/moonlink/src/storage/wal.rs

```diff
@@ -442,14 +536,29 @@ impl WalManager {
     }
 
     /// Should be called after a persist and truncate operation has completed. Updates
-    /// tracked files and transactions.
-    pub fn handle_completed_persist_and_truncate(
+    /// tracked files and transactions. Returns the highest LSN that has been persisted into WAL.
+    pub fn handle_completed_persistence_and_truncate(
         // For now, we handle the persist and truncate results together.
         &mut self,
         persist_and_truncate_result: &PersistAndTruncateResult,
-    ) {
+    ) -> Option<u64> {
         self.handle_complete_persistence(persist_and_truncate_result);
         self.handle_complete_truncate(persist_and_truncate_result);
+
+        persist_and_truncate_result
+            .file_persisted
+            .as_ref()
+            .map(|wal_file_info| wal_file_info.highest_lsn)
+    }
+
+    // ------------------------------
+    // Drop WAL files
+    // ------------------------------
+    pub async fn drop_wal(&mut self) -> Result<()> {
```

Comment:
if you put everything under a certain directory (current impl is write-through cache directory), why not delete the whole directory? it could also clean up possible orphaned files